jdk 9 compilation fails

```
[INFO] Compiling 25 source files to /Users/svaddi/SreeVaddi/source/github/sreev/bdg-formats/target/classes
[INFO] -------------------------------------------------------------
[ERROR] COMPILATION ERROR :
[INFO] -------------------------------------------------------------
[ERROR] Source option 1.5 is no longer supported. Use 1.6 or later.
[ERROR] Target option 1.5 is no longer supported. Use 1.6 or later.
[INFO] 2 errors
[INFO] -------------------------------------------------------------
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 4.232 s
[INFO] Finished at: 2018-05-19T07:42:19-07:00
[INFO] ------------------------------------------------------------------------

```

--- a/pom.xml
+++ b/pom.xml
@@ -54,6 +54,7 @@
 
   <properties>
     <avro.version>1.8.1</avro.version>
+    <java.version>1.8</java.version>
   </properties>
   
   <build>
@@ -66,8 +67,17 @@
         </plugin>
         <plugin>
           <groupId>org.apache.maven.plugins</groupId>
+          <artifactId>maven-compiler-plugin</artifactId>
+          <version>3.8.0</version>
+          <configuration>
+            <source>${java.version}</source>
+            <target>${java.version}</target>
+          </configuration>
+        </plugin>
+        <plugin>
+          <groupId>org.apache.maven.plugins</groupId>
           <artifactId>maven-javadoc-plugin</artifactId>
-          <version>2.10.3</version>
+          <version>3.0.1</version>
           <configuration>
             <additionalparam>-Xdoclint:none</additionalparam>
           </configuration>
@@ -96,7 +106,24 @@
       </plugin>
     </plugins>
   </build>
-  
+
+  <dependencyManagement>
+    <dependencies>
+      <dependency>
+        <groupId>org.apache.avro</groupId>
+        <artifactId>avro</artifactId>
+        <version>${avro.version}</version>
+      </dependency>
+    </dependencies>
+  </dependencyManagement>
+  <dependencies>
+    <dependency>
+      <groupId>org.apache.avro</groupId>
+      <artifactId>avro</artifactId>
+      <scope>compile</scope>
+    </dependency>
+  </dependencies>
+
   <profiles>
     <!-- Only sign artifacts when we are performing a release, not snapshots -->
     <profile>
@@ -106,7 +133,7 @@
           <plugin>
             <groupId>org.apache.maven.plugins</groupId>
             <artifactId>maven-source-plugin</artifactId>
-            <version>3.0.0</version>
+            <version>3.0.1</version>
             <executions>
               <execution>
                 <id>attach-sources</id>
@@ -119,7 +146,7 @@
           <plugin>
             <groupId>org.apache.maven.plugins</groupId>
             <artifactId>maven-javadoc-plugin</artifactId>
-            <version>2.10.3</version>
+            <version>3.0.1</version>
             <configuration>
               <additionalparam>-Xdoclint:none</additionalparam>
             </configuration>
@@ -149,56 +176,5 @@
         </plugins>
       </build>
     </profile>
-
-    <profile>
-      <!--
-        Insert UML diagrams within the HTML generated Javadoc.
-
-        The UmlGraphDoc requires GraphViz to be installed in the classpath of the target system
-        (more specifically the 'dot' executable)
-        
-        USAGE
-        
-          mvn javadoc:javadoc -P uml-diagrams
-
-        DOCUMENTATION
-          
-          * Parameters (in additionalparam section below)
-          http://www.umlgraph.org/doc/cd-opt.html
-      -->
-      <id>uml-diagrams</id>
-      <build>
-        <plugins>
-          <plugin>
-            <artifactId>maven-javadoc-plugin</artifactId>
-            <version>2.9.1</version>
-            <configuration>
-              <aggregate>true</aggregate>
-              <show>private</show>
-              <doclet>org.umlgraph.doclet.UmlGraphDoc</doclet>
-              <docletArtifact>
-                <groupId>org.umlgraph</groupId>
-                <artifactId>umlgraph</artifactId>
-                <version>5.6.6</version>
-              </docletArtifact>
-              <additionalparam>
-                -attributes  -enumerations  -enumconstants  -qualify  -types  -visibility
-                -inferdep  -inferrel  -collpackages java.util.*
-                -hide java.*  -hide org.bdgenomics.formats.avro.*.Builder
-                -postfixpackage  -nodefontsize 9  -nodefontpackagesize 7
-              </additionalparam>
-            </configuration>
-          </plugin>
-        </plugins>
-      </build>
-    </profile>
   </profiles>
-
-  <dependencies>
-    <dependency>
-      <groupId>org.apache.avro</groupId>
-      <artifactId>avro</artifactId>
-      <version>${avro.version}</version>
-    </dependency>
-  </dependencies>
 </project>

